src/libfreenect2.cpp

```diff
@@ -1115,4 +1119,239 @@ Freenect2Device *Freenect2::openDefaultDevice(const PacketPipeline *pipeline)
+{
+  ir_camera_params_ = params;
+  DepthPacketProcessor *proc = pipeline_->getDepthPacketProcessor();
+  if (proc != 0)
+  {
+    IrCameraTables tables(params);
+    proc->loadXZTables(&tables.xtable[0], &tables.ztable[0]);
+    proc->loadLookupTable(&tables.lut[0]);
+  }
+}
+
+void Freenect2ReplayDevice::setConfiguration(const Freenect2Device::Config &config)
+{
+  DepthPacketProcessor *proc = pipeline_->getDepthPacketProcessor();
+  if (proc != 0)
+    proc->setConfiguration(config);
+}
+
+void Freenect2ReplayDevice::setColorFrameListener(FrameListener* listener)
+{
+  RgbPacketProcessor* proc = pipeline_->getRgbPacketProcessor();
+  if (proc != NULL)
+  {
+    proc->setFrameListener(listener);
+  }
+}
+
+void Freenect2ReplayDevice::setIrAndDepthFrameListener(FrameListener* listener)
+{
+  DepthPacketProcessor* proc = pipeline_->getDepthPacketProcessor();
+  if (proc != NULL)
+  {
+    proc->setFrameListener(listener);
+  }
+}
+
+bool Freenect2ReplayDevice::processRawFrame(Frame::Type type, Frame* frame)
+{
+  if (frame->format != Frame::Raw)
+  {
+    return false;
+  }
+  switch (type)
+  {
+  case Frame::Color:
+    processRgbFrame(frame);
+    break;
+  case Frame::Depth:
+    processDepthFrame(frame);
+    break;
+  default:
+    return false;
+  }
+  return true;
+}
+
+void Freenect2ReplayDevice::processRgbFrame(Frame* frame)
+{
+  RgbPacket packet;
+  packet.timestamp = frame->timestamp;
+  packet.sequence = frame->sequence;
+  packet.jpeg_buffer = frame->data;
+  packet.jpeg_buffer_length = frame->bytes_per_pixel;
+  packet.exposure = frame->exposure;
+  packet.gain = frame->gain;
+  packet.gamma = frame->gamma;
+
+  pipeline_->getRgbPacketProcessor()->process(packet);
+}
+
+void Freenect2ReplayDevice::processDepthFrame(Frame* frame)
+{
+  DepthPacket packet;
+  packet.timestamp = frame->timestamp;
+  packet.sequence = frame->sequence;
+  packet.buffer = frame->data;
+  packet.buffer_length = frame->bytes_per_pixel;
+ 
+  pipeline_->getDepthPacketProcessor()->process(packet);
+}
+
+void Freenect2ReplayDevice::loadP0Tables(unsigned char* buffer, size_t buffer_length)
+{
+  pipeline_->getDepthPacketProcessor()->loadP0TablesFromCommandResponse(buffer, buffer_length);
+}
+
+void Freenect2ReplayDevice::loadXZTables(const float *xtable, const float *ztable)
+{
+  pipeline_->getDepthPacketProcessor()->loadXZTables(xtable, ztable);
+}
+
+void Freenect2ReplayDevice::loadLookupTable(const short *lut)
+{
+  pipeline_->getDepthPacketProcessor()->loadLookupTable(lut);
+}
+
+void runner (void* arg)
+{
+  static_cast<Freenect2ReplayDevice*>(arg)->run();
+}
+
+bool Freenect2ReplayDevice::start()
+{
+  running_ = true;
+  startStreams(true, true);
+  t_ = new thread(runner, this);
+  return running_;
+}
+
+bool Freenect2ReplayDevice::startStreams(bool enable_rgb, bool enable_depth)
+{
+  LOG_INFO << "starting...: rgb: " << enable_rgb << ", depth: " << enable_depth;
```

Comment:
Return false if you do not actually implement this function.

Reply:
Seems odd, but OK.

Reply:
DONE